src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -147,13 +147,21 @@ internal async Task InitializeOnActivatedAsync(CancellationToken cancellationTok
                 _packageSources.CurrentChanged += OnSelectedPackageSourceChanged;
                 PackageSourcesListBox.GotFocus += PackageSourcesListBox_GotFocus;
                 PackageSourcesListBox.DataSource = _packageSources;
+                for (int i = 0; i < packageSources.Count; i++)
```

Comment:
Why doesn't the binding work here?

Reply:
I see. 

The thing that threw me off is why we need to this now. 
I imagine this is what sets the checkbox state right?

Reply:
NuGet does custom rendering of the items in the list, and historically we've rendered the "not really bound" object's state, not the checked list box's internal state. See PackageSourceCheckedListBox.cs line 65.  These two for loops synchronize the lists's internal checkbox state with our model's state.